### ISSUE TITLE
limit the number of threads

### ISSUE THREAD

**Issue by alexweisberg** [zhangyumeng1sjtu/T4SEfinder#2]
Hi, T4SEfinder is great and very useful for my research. I am looking to run it on a computing cluster, however by default it tries to use all of the available CPUs in the environment. Could you please add a threads option so that I can set a limit for the maximum number of CPUs to use? Thank you!

**Comment by zhangyumeng1sjtu** [comment]
T4SEfinder uses a single CPU  by default when I test on our computing cluster. Could you provide some detailed running information to help improve the program? Thank you!

**Comment by alexweisberg** [comment]
I installed T4SEfinder by cloning the github repo and then installing the prerequisites with pip3.7 install T4SEfinder.

The machine I am running this on has 64 CPUs/processors. It does not have a GPU (hence the CUDA warning).

Here is what I get when I run it with a test dataset:

`python3 main.py -in AS1D4test.faa -weights weights/mlp/ -vote tapebert_mlp`
`/homefolder/.local/lib/python3.7/site-packages/torch/cuda/__init__.py:52: UserWarning: CUDA initialization: Found no NVIDIA driver on your system. Please check that you have an NVIDIA GPU and installed a driver from http://www.nvidia.com/Download/index.aspx (Triggered internally at  /pytorch/c10/cuda/CUDAFunctions.cpp:100.)
  return torch._C._cuda_getDeviceCount() > 0
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 461/461 [03:09<00:00,  2.40it/s]`

The program produces output in the results folder that looks correct. 

When I view the running processes on that machine using htop, the program has spawned many threads:
<img width="1676" alt="Screenshot 2023-02-05 at 5 23 51 PM" src="https://user-images.githubusercontent.com/12143265/216861769-7a742260-f0ad-44d5-817e-3c9aad751627.png">

**Comment by alexweisberg** [comment]
As a followup to this, when I add the statement:
` print("number of torch cpus: " + str(torch.get_num_threads()))`
to main.py I get the output:
`number of torch cpus: 32`

So it appears to be torch that is assuming to use all processors. 

If the program is going to use CPU instead of GPU, it would be helpful to have a threads option that is passed to torch via torch.set_num_threads so that it only uses the specified number of processors. I think it was only using 1 CPU on your machine because your computer has a GPU.

Thanks!

**Comment by zhangyumeng1sjtu** [comment]
Yes. This problem occurs in the process of `tape-embed`(https://github.com/songlab-cal/tape) if GPU is not available. Unfortunately, adding `torch.set_num_thread()` in `main.py` cannot limit the number of threads when generating the protein embeddings. It is recommended to modify the source code of `tape-embed` or try the online service(https://tool2-mml.sjtu.edu.cn/T4SEfinder_TAPE/).
